proxy/http2/Http2ConnectionState.cc

```diff
@@ -857,6 +857,7 @@ Http2ConnectionState::main_event_handler(int event, void *edata)
 
   // Finalize HTTP/2 Connection
   case HTTP2_SESSION_EVENT_FINI: {
+    SCOPED_MUTEX_LOCK(lock, this->mutex, this_ethread());
```

Comment:
@sidhuagarwal Added `SCOPED_MUTEX_LOCK(lock, this->ua_session->mutex, this_ethread());` in `cleanup_streams()` and `release_stream()`. Could you verify the crash doesn't happen?